e2e/scenario/validators.go

```diff
@@ -103,6 +103,45 @@ func NonEmptyDirectoryValidator(dirName string) *LiveVMValidator {
 	}
 }
 
+func FileHasContentsValidator(fileName string, contents string) *LiveVMValidator {
+	return &LiveVMValidator{
+		Description: fmt.Sprintf("Assert that %s has defined contents", fileName),
+		Command:     fmt.Sprintf("cat %s | grep -q '%s'", fileName, contents),
+		Asserter: func(code, stdout, stderr string) error {
+			if code != "0" {
+				return fmt.Errorf("expected to find a file '%s' with contents '%s' but did not", fileName, contents)
+			}
+			return nil
+		},
+	}
+}
+
+func serviceCanRestartValidator(serviceName string, timeoutInSeconds int) *LiveVMValidator {
+	steps := []string{
+		"echo Existing processes:",
+		fmt.Sprintf("pgrep --list-full %s", serviceName),
+		"echo Killing...",
+		// we use systemctl kill rather than kill -9 because container restrictions stop us sending a kill sig to a process
+		fmt.Sprintf("systemctl kill %s", serviceName),
+		"echo Sleeping...",
+		fmt.Sprintf("sleep %d", timeoutInSeconds),
+		"echo Checking processes have restarted...",
+		fmt.Sprintf("pgrep --list-full %s", serviceName),
```

Comment:
we should ensure that is has a different PID

Reply:
That's a good idea. I was having issues using "pgrep" in this particular command line and have since changed to a different approach. Once I've got these tests actually working, I'll try to add the pid change check. Or just a check that it's inactive after killing it.

Reply:
Added this check.